openstack/check.py

```diff
@@ -448,7 +449,16 @@ def get_network_stats(self):
 
         # FIXME: (aaditya) Check all networks defaults to true until we can reliably assign agents to networks to monitor
         if self.init_config.get('check_all_networks', True):
-            network_ids = list(set(self.get_all_network_ids()) - set(self.init_config.get('exclude_network_ids', [])))
+            all_network_ids = set(self.get_all_network_ids())
+            exclude_network_ids = set(
+                [re.compile(ex) for ex in self.init_config.get('exclude_network_ids', [])]
```

Comment:
I think you probably want to compile the regex rules just once (`__init__()`), and only do the filtering here.